File: HackerRank Solutions/Algorithms/Implementation/Migratory Birds.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

public class Solution {

    // Complete the migratoryBirds function below.
    static int migratoryBirds(List<Integer> arr) {
        int[] frequencyArray = new int[5];
        for (Integer i : arr) {
            frequencyArray[i-1] += 1;
        }
        int maxFrequency = frequencyArray[0];
        int maxFrequencyBird = 1;
        for (int i = 1; i < frequencyArray.length; i++) {
            if (frequencyArray[i] > maxFrequency) {
                maxFrequency = frequencyArray[i];
                maxFrequencyBird = i+1;
            }
        }
        return maxFrequencyBird;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int arrCount = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        int result = migratoryBirds(arr);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: HackerRank Solutions/Algorithms/Implementation/Migratory Birds.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the migratoryBirds function below.
def migratoryBirds(arr):
    frequencyArray = [0] * 5
    for i in range(len(arr)):
        frequencyArray[arr[i]-1] += 1
    maxFrequency = frequencyArray[0]
    maxFrequencyBird = 1
    for i in range(1, len(frequencyArray)):
        if frequencyArray[i] > maxFrequency:
            maxFrequency = frequencyArray[i]
            maxFrequencyBird = i+1
    return maxFrequencyBird

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    arr_count = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    result = migratoryBirds(arr)

    fptr.write(str(result) + '\n')

    fptr.close()
